### PR TITLE
[xcode11] Fix several issues to make the mtouch tests pass. Fixes #6259.

### LINKED ISSUE #6259: [xcode11] Numerous mtouch test failures
https://jenkins.mono-project.com/job/xamarin-macios-pr-builder/10186/Test_20Report/

[Test Count: 426, Passed: 358, Failed: 63, Warnings: 0, Inconclusive: 0, Skipped: 5](https://jenkins.mono-project.com/job/xamarin-macios-pr-builder/10186/Test_20Report/MTouch%20tests/451/execute-20190605_204222.txt)

### PR DESCRIPTION
**Pull request by rolfbjarne**
* [msbuild] 'metal' and 'metallib' aren't in PATH anymore, so use xcrun to execute them.
* [tests] Adjust tests for changes in ld warning format.
* [tests] Adjust tests for name changes in Clang.
* [tests] Adjust tests for changes in 'nm' output.
* [tests] Adjust min OS version tests for Xcode 11b1.
* [tests] Ignore all 'MT0099: Not linking with WatchKit because Xcode 11 beta 1' warnings in tests.
* [tests] Minor refactor to get better Xcode version parsing.

Fixes https://github.com/xamarin/xamarin-macios/issues/6259.

### CODE REVIEW

File: tests/common/Configuration.cs [hunk 1]
```diff
@@ -43,6 +43,15 @@ static partial class Configuration
 		public static bool include_watchos;
 		public static bool include_device;
 
+		static Version xcode_version;
+		public static Version XcodeVersion {
+			get {
+				if (xcode_version == null)
+					xcode_version = Version.Parse (XcodeVersionString);
```
Comment (spouliot) [inline tests/common/Configuration.cs]: `TryParse` and log failures ?

File: tests/common/Configuration.cs [hunk 2]
```diff
@@ -75,7 +84,7 @@ public static string GetOldXcodeRoot (Version min_version = null)
 			var xcodes = Directory.GetDirectories ("/Applications", "Xcode*.app", SearchOption.TopDirectoryOnly);
 			var with_versions = new List<Tuple<Version, string>> ();
 
-			var max_version = Version.Parse (XcodeVersion);
+			var max_version = Version.Parse (XcodeVersionString);
```
Comment (spouliot) [inline tests/common/Configuration.cs]: `TryParse` and log failures ?

File: tests/common/Configuration.cs [hunk 3]
```diff
@@ -543,7 +552,7 @@ public static void AssertXcodeSupports32Bit ()
 
 		public static bool XcodeSupports32Bit {
 			get {
-				return Version.Parse (XcodeVersion).Major < 10;
+				return Version.Parse (XcodeVersionString).Major < 10;
```
Comment (spouliot) [inline tests/common/Configuration.cs]: can't it re-use the already parsed version string ?

File: tests/common/ExecutionHelper.cs [hunk 4]
```diff
@@ -134,6 +134,12 @@ public void ParseMessages ()
 				var line = l;
 				var msg = new ToolMessage ();
 				var origin = string.Empty;
+
+				if (l.Contains ("warning MT0099: Not linking with WatchKit because Xcode 11 beta 1 does not ship with the WatchKit framework")) {
```
Comment (spouliot) [inline tests/common/ExecutionHelper.cs]: comment / FIXME to make reverting this easier

File: tests/common/Configuration.cs (hunk 3 above)
Comment (rolfbjarne) [reply tests/common/Configuration.cs]: Yes, let me change that.

File: tests/common/Configuration.cs (hunk 1 above)
Comment (rolfbjarne) [reply tests/common/Configuration.cs]: If we fail to parse the Xcode version things are going sideways pretty quickly anyway.

File: tests/common/ExecutionHelper.cs (hunk 4 above)
Comment (rolfbjarne) [reply tests/common/ExecutionHelper.cs]: It's already listed in #6212, but I'll add some text here too.

File: tests/common/Configuration.cs (hunk 1 above)
Comment (spouliot) [reply tests/common/Configuration.cs]: true, but will it be easy to diagnose for a customer reporting the issue ?

Comment (rolfbjarne) [reply tests/common/Configuration.cs]: This is in tests/, so it's not customer-facing.

**Comment by rolfbjarne** [comment]
@spouliot 

> using `xcrun` -> #4634 (partial)

Using `xcrun` didn't work everywhere, so I came up with a different solution. Everything else should be fixed now.

File: tests/common/ExecutionHelper.cs [hunk 5]
```diff
@@ -134,6 +134,13 @@ public void ParseMessages ()
 				var line = l;
 				var msg = new ToolMessage ();
 				var origin = string.Empty;
+
+				// FIXME: remove this warning check once WatchKit is back with us again.
+				if (l.Contains ("warning MT0099: Not linking with WatchKit because Xcode 11 beta 1 does not ship with the WatchKit framework")) {
```
Comment (chamons) [inline tests/common/ExecutionHelper.cs]: Beta2 now?

File: tests/mtouch/MTouch.cs [hunk 6]
```diff
@@ -3995,6 +3999,8 @@ public void SimlauncherSymbols (string arch, string simlauncher_suffix)
 				case "____chkstk_darwin": // compiler magic, unrelated to XI/XM
 				case "___block_descriptor_28_e5_v4@?0l": // new Xcode 10.2 clang option
 				case "___block_descriptor_48_e5_v8@?0l": // new Xcode 10.2 clang option
+				case "___block_descriptor_28_e5_v4\u0001?0l": // Xcode 11 b1 name
```
Comment (chamons) [inline tests/mtouch/MTouch.cs]: b2?

**Comment by rolfbjarne** [comment]
Test failures are unrelated

* xammac tests/Mac Unified/Debug: https://github.com/mono/mono/issues/14975
* mono-native-compat/Mac Unified/Modern: https://github.com/xamarin/xamarin-macios/issues/6322
* mono-native-unified/Mac Unified/Modern: https://github.com/xamarin/xamarin-macios/issues/6322
* BCL tests: https://github.com/xamarin/xamarin-macios/issues/6261
* MSBuild tests/iOS: https://github.com/xamarin/xamarin-macios/issues/6325
* mmptest/macOS/NonSystem: https://github.com/xamarin/xamarin-macios/issues/6257

**Pull request merged**